src/lm/diff.cpp

```diff
@@ -0,0 +1,255 @@
+/**
+ * @file diff.cpp
+ * @author Sean Massung
+ */
+
+#include <algorithm>
+#include <queue>
+#include "lm/diff.h"
+#include "porter2_stemmer.h"
+
+namespace meta
+{
+namespace lm
+{
+diff::diff(const cpptoml::table& config) : lm_{config}
+{
+    auto table = config.get_table("diff");
+    if (!table)
+        throw diff_exception{"missing [diff] table from config"};
+
+    auto nval = table->get_as<int64_t>("n-value");
+    if (!nval)
+        throw diff_exception{"n-value not specified in config"};
+    n_val_ = *nval;
+
+    auto edits = table->get_as<int64_t>("max-edits");
+    if (!edits)
+        throw diff_exception{"max-edits not specified in config"};
+    max_edits_ = *edits;
+
+    auto b_pen = table->get_as<double>("base-penalty");
+    base_penalty_ = b_pen ? *b_pen : 0.0;
```

Comment:
I should probably add `value_or(T)` to `cpptoml::option<T>` (and `meta::util::optional<T>`)...